src/oscar/templates/oscar/catalogue/browse.html

```diff
@@ -64,7 +64,7 @@
         {% for value in selected_facets %}
             <input type="hidden" name="selected_facets" value="{{ value }}" />
         {% endfor %}
-        <input type="hidden" name="q" value="{{ search_form.q.value }}" />
+        <input type="hidden" name="q" value="{% firstof search_form.q.value %}" />
```

Comment:
Yes, `default_if_none` is surely more appropriate here. Updated.

The web ui doesn't let me squash commits. Hope that's ok.